### DISCUSSION TITLE
DNSCrypt gives up if it can't immediately download the list of proxies

### DISCUSSION THREAD

**Discussion by iopq** [DNSCrypt/dnscrypt-proxy#1991] [Potential issues]
Since I am behind the GFW, I need it to first resolve my proxy's domain name and then try again to download the list

But it doesn't do this, it dies too quickly, so my proxy can't connect either

**Comment by jedisct1** [comment]
If it doesn't have any servers list, there's no way to connect to any servers :)

Maybe what you can do is add your proxy's domain name to the set of cloaking rules (to hard-code its IP address), or to the forwarding rules (so the domain is resolved by another resolver).

**Reply by iopq** [reply]
> If it doesn't have any servers list, there's no way to connect to any servers :)
> 
> Maybe what you can do is add your proxy's domain name to the set of cloaking rules (to hard-code its IP address), or to the forwarding rules (so the domain is resolved by another resolver).

But it already downloaded the proxy list like an hour ago, it just dies immediately if it can't update. That can't be the best default behavior, it should be more resilient and first try the old list before updating and comparing the changes (if there are new servers to test)
